api/envoy/config/core/v3/address.proto

```diff
@@ -32,15 +32,19 @@ message Pipe {
 }
 
 // The address represents an envoy internal listener.
-// [#comment: TODO(lambdai): Make this address available for listener and endpoint.
 // TODO(asraa): When address available, remove workaround from test/server/server_fuzz_test.cc:30.]
 message EnvoyInternalAddress {
   oneof address_name_specifier {
     option (validate.required) = true;
 
-    // [#not-implemented-hide:] The :ref:`listener name <envoy_v3_api_field_config.listener.v3.Listener.name>` of the destination internal listener.
+    // Specifies the :ref:`name <envoy_v3_api_field_config.listener.v3.Listener.name>` of the
+    // internal listener.
     string server_listener_name = 1;
   }
+
+  // Specifies an endpoint identifier to distinguish between multiple endpoints for the same internal listener in a
+  // single upstream pool.
+  string endpoint_id = 2;
```

Comment:
Is this kind of multiple internal addresses? if yes, then another option is just to support multiple internal addresses here, then you needn't add `endpoint_id` field for multiple endpoints, just specify multiple internal addresses. but yes, you can't limit the internal address to the listener name anymore.

Reply:
after looking at another PR, multiple addresses will bind the number of listener addresses and the number of endpoints together, I guess that isn't what people want.

Reply:
Correct, we want to maximize reuse of the internal listeners - e.g. tunneling shouldn't be coupled to the origin of the traffic. So that implies a possibility of multiple endpoints to the same internal listener in one pool. Upstream de-duplicates endpoints by computing `asString` so it breaks there.

Reply:
Is there any typical value of endpoint_id? e.g. the actually L4 address?

Reply:
@lambdai yes exactly

Reply:
I think it'd be confusing to say "L4 address should be here". We don't say a listener corresponds to VIP, or a cluster corresponds to a VIP. That's too prescriptive. I will add an example of the usage.